docs/docs/sourcing-from-the-filesystem.md

```diff
@@ -2,7 +2,105 @@
 title: Sourcing from the Filesystem
 ---
 
-This is a stub. Help our community expand it.
+This guide will walk you through sourcing from the filesystem.
 
-Please use the [Gatsby Style Guide](/docs/gatsby-style-guide/) to ensure your
-pull request gets accepted.
+## Setup
+
+This guide assumes that you have a Gatsby project set up. If you need to set up a project, please reference the [Quick Start Guide](https://github.com/gatsbyjs/gatsby/tree/master/docs). In addition, it will be useful if you are familiar with Graph_i_QL, a tool that helps you structure your queries correctly.
+
+## Introducing Graph_i_QL
```

Comment:
I don't really think this section is necessary for this doc. I'll explain further in a different comment in this review.